From 0058cd167e79e115c2fba1835247961a476b390a Mon Sep 17 00:00:00 2001
From: "Satyanarayana, Sandhya" <sandhya.satyanarayana@ti.com>
Date: Thu, 12 Jul 2012 13:16:12 +0530
Subject: [PATCH 1049/1092] ARM: OMAP: AM33XX: PM: Fix build error on
 CONFIG_SUSPEND disable

On disabling CONFIG_SUSPEND, error and warnings were seen
during kernel build.

This patch fixes them.

Signed-off-by: Satyanarayana, Sandhya <sandhya.satyanarayana@ti.com>
---
 arch/arm/mach-omap2/pm33xx.c | 4 +++-
 arch/arm/mach-omap2/pm33xx.h | 2 ++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-omap2/pm33xx.c b/arch/arm/mach-omap2/pm33xx.c
index d2edba0..34d5366 100644
--- a/arch/arm/mach-omap2/pm33xx.c
+++ b/arch/arm/mach-omap2/pm33xx.c
@@ -490,15 +490,18 @@ void am33xx_push_sram_idle(void)
 static int __init am33xx_pm_init(void)
 {
 	int ret;
+#ifdef CONFIG_SUSPEND
 	void __iomem *base;
 	u32 reg;
 	u32 evm_id;
 
+#endif
 	if (!cpu_is_am33xx())
 		return -ENODEV;
 
 	pr_info("Power Management for AM33XX family\n");
 
+#ifdef CONFIG_SUSPEND
 /* Read SDRAM_CONFIG register to determine Memory Type */
 	base = am33xx_get_ram_base();
 	reg = readl(base + EMIF4_0_SDRAM_CONFIG);
@@ -522,7 +525,6 @@ static int __init am33xx_pm_init(void)
 	else
 		suspend_cfg_param_list[EVM_ID] = 0xff;
 
-#ifdef CONFIG_SUSPEND
 	(void) clkdm_for_each(clkdms_setup, NULL);
 
 	/* CEFUSE domain should be turned off post bootup */
diff --git a/arch/arm/mach-omap2/pm33xx.h b/arch/arm/mach-omap2/pm33xx.h
index 5fcb0f5..adf56bf 100644
--- a/arch/arm/mach-omap2/pm33xx.h
+++ b/arch/arm/mach-omap2/pm33xx.h
@@ -45,6 +45,7 @@ struct am33xx_padconf_regs {
 	u32 val;
 };
 
+#ifdef CONFIG_SUSPEND
 static struct am33xx_padconf_regs am33xx_lp_padconf[] = {
 	{.offset = AM33XX_CONTROL_GMII_SEL_OFFSET},
 	{.offset = AM33XX_CONTROL_PADCONF_GPMC_A0_OFFSET},
@@ -81,6 +82,7 @@ static struct am33xx_padconf_regs am33xx_lp_padconf[] = {
 	{.offset = AM33XX_CONTROL_PADCONF_MDIO_DATA_OFFSET},
 	{.offset = AM33XX_CONTROL_PADCONF_MDIO_CLK_OFFSET},
 };
+#endif /* CONFIG_SUSPEND */
 #endif /* ASSEMBLER */
 
 #define M3_TXEV_EOI			(AM33XX_CTRL_BASE + 0x1324)
-- 
1.7.11.2

